### DISCUSSION TITLE
I want to have time range options so that I can check "Trend" for a certain time.

### DISCUSSION THREAD

**Discussion by elainetao829** [allure-framework/allure2#1466] [Ideas & Feedback]
**Is your feature request related to a problem? Please describe.**
I want to send results and generate reports after every execution. But in that case, The Trend will be really complicated with too much information. Sometimes, the project wants only the Trend for a certain period.

**Describe the solution you'd like**
Have something like a time bar, so that users can select a certain time period to see the Trend during that time. 

**Describe alternatives you've considered**
Clear historic data manually every time and generate reports again for executions during the time period that we want to check. But that sounds like a lot of manual work.

**Comment by elainetao829** [comment]
Hi, 
Any comments on this?

**Reply by baev** [reply]
you can get much more control over your tests history data while using Allure TestOps.

**Reply by elainetao829** [reply]
But is there any way that Allure can support it by itself? At the moment, we already paid for a test case management tool. So I do not see the need to buy another tool (even it's maybe better). We are not a small project but a large company. Switching from one tool to another means every tester in our company needs to move from A to B, which is not that easy :(

**Reply by baev** [reply]
> But is there any way that Allure can support it by itself?

Allure Report will always have limitations in history related features due to lack of database. It's designed to answer the only question: what is happed in this particular test run. History, trends and analytics are very simple - basically we enrich each report with the links to the other report if you provide such data during report generation.

We can try to implement some additional features like this one, but it will make report itself much more complicated. At we like to keep it lightweight and as simple as possible.  

From our experience at some point you can't make any further without having some centralised test reporting solution. And Allure TestOps is the best one on the market. It is designed to show the overall picture of your testing. 

> At the moment, we already paid for a test case management tool. So I do not see the need to buy another tool (even it's maybe better). We are not a small project but a large company. Switching from one tool to another means every tester in our company needs to move from A to B, which is not that easy :(

https://qameta.io/blog/top-5-allure-testops-myths/
https://qameta.io/blog/wisebits-allure-testops-story/
